### DISCUSSION TITLE
UDP benchmark support

### DISCUSSION THREAD

**Discussion by petrih479** [openthread/openthread#8716] [Ideas]
Hello,
Thread:CLI sample supports command ot tcp benchmark. 

I would appreciate similar command supporting UDP, like ot udp benchmark.

I have tried include zperf in the build, but I don't get it work as expected. More closely, it doesn't succeed measure data throughput.

Best Regards,
Petri

**Comment by jwhui** [comment]
Have you consulted the Zephyr community regarding zperf?

When using OpenThread with a Linux host, you could use [iPerf](https://iperf.fr/).

When using something like `iperf` to test network performance with UDP, you should understand its limitations. In particular, because UDP does not implement things like congestion control or flow control, you are basically measuring the network's ability to buffer messages.

**Reply by jchellap** [reply]
Hi @jwhui ,

Do we have any theoretical formula for UDP throughput for open thread? This is to understand what is the expected value for the given bandwidth and data length

**Reply by jwhui** [reply]
Unfortunately, I don't have anything readily available. Note that because Thread implements 6LoWPAN header compression, that the header overhead will depend on the nature of the traffic itself.

**Comment by petrih479** [comment]
Hello jwhui,
Yes, I tried to implement zperf in thread:cli sample, but couldn't get it work. One proposal from tech support was to create feature request which I did. 
I want to do simple throughput test directly on console without need for external linux host. 
Yes, I'm aware of the benefits of the UDP over TCP when measuring real data throughput on poor wireless links, i.e. no ping-ponging of packets due to non-acknowledged packet in TCP.
BR Petri

**Reply by jwhui** [reply]
> Yes, I'm aware of the benefits of the UDP over TCP when measuring real data throughput on poor wireless links, i.e. no ping-ponging of packets due to non-acknowledged packet in TCP.

My point is that raw UDP throughput is not necessarily a useful metric.
